src/main/kotlin/apoc/nlp/gcp/GCPProcedures.kt

```diff
@@ -0,0 +1,176 @@
+package apoc.nlp.gcp
+
+import apoc.ai.dto.AIMapResult
+import apoc.graph.document.builder.DocumentToGraph
+import apoc.graph.util.GraphsConfig
+import apoc.result.RelationshipResult
+import apoc.result.VirtualGraph
+import apoc.result.VirtualNode
+import apoc.util.Util
+import org.neo4j.graphdb.Node
+import org.neo4j.graphdb.Relationship
+import org.neo4j.graphdb.RelationshipType
+import org.neo4j.graphdb.Transaction
+import org.neo4j.logging.Log
+import org.neo4j.procedure.*
+import java.util.stream.Stream
+
+class GCPProcedures {
+    @Context
+    @JvmField
+    var log: Log? = null
+
+    @Context
+    @JvmField
+    var tx: Transaction? = null
+
+    @Procedure(value = "apoc.nlp.gcp.entities.stream", mode = Mode.READ)
+    @Description("Returns a stream of entities for provided text")
+    fun entitiesStream(@Name("sourceNode") sourceNode: Node,
+                       @Name(value = "config", defaultValue = "{}") config: Map<String, Any>)
+            : Stream<AIMapResult> {
+        verifyKey(config, "key")
+        val nodeProperty = getNodeProperty(config)
+        verifyNodeProperty(sourceNode, nodeProperty)
+
+        return Stream.of(entities(config, sourceNode, nodeProperty))
+    }
+
+    @Procedure(value = "apoc.nlp.gcp.entities.graph", mode = Mode.WRITE)
+    @Description("Creates a (virtual) entity graph for provided text")
+    fun entitiesGraph(@Name("sourceNode") sourceNode: Node,
+                      @Name(value = "config", defaultValue = "{}") config: Map<String, Any>)
+            : Stream<VirtualGraph> {
+        verifyKey(config, "key")
+        val nodeProperty = getNodeProperty(config)
+        verifyNodeProperty(sourceNode, nodeProperty)
+
+        val response = entities(config, sourceNode, nodeProperty).response
+
+        val storeGraph:Boolean = config.getOrDefault("write", false) as Boolean
+        val graphConfig = mapOf(
+                "skipValidation" to true,
+                "mappings" to mapOf("$" to "Entity{!name,type,@metadata}"),
+                "write" to storeGraph
+        )
+
+        val documentToGraph = DocumentToGraph(tx, GraphsConfig(graphConfig))
+        val graph = documentToGraph.create(response["entities"])
+
+        val mutableGraph = graph.graph.toMutableMap()
+
+        val nodes = (mutableGraph["nodes"] as Set<Node>).toMutableSet()
+        val relationships = (mutableGraph["relationships"] as Set<Relationship>).toMutableSet()
+        if(storeGraph) {
```

Comment:
we can use a functional way to do this:
```suggestion
        val node = if(storeGraph) {
            mergeRelationships(sourceNode, nodes, classifyRelationshipType(config)).forEach { rel -> relationships.add(rel) }
            sourceNode
        } else {
            val virtualNode = VirtualNode(sourceNode, sourceNode.propertyKeys.toList())
            createRelationships(virtualNode, nodes, classifyRelationshipType(config)).forEach { rel -> relationships.add(rel) }
            virtualNode
        }
        nodes.add(node)
```